keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1229 @@
+A node should be able to allow both types of cores, without having to have one setting
+or another, and be able to pull from these pulls accordingly.  Additionally, there may 
+be a need to have some high-priority cores for higher performance and other lower-priority
+cores for other less-sensitive parts of a workloads.  In these use cases, the workloads 
+may also have particular types of NUMA splits required.
+
+#### Power optimization of workloads
+
+Cores should be able to be quickly spun up or down according to resource requirements.
+Additionally, the nodes should be as densely packed as possible regrading the abliity to
+do core use.  There should also be the ability to choose between efficiency cores and
+performance cores within newer architectures, according to workload requirements.
+
+#### Research of new resource management patterns within the cloud
+
+There are a variety of modifiers that can be placed around cores.  Static cores,
+isolated cores, shared cores, efficiency cores, and performance cores are only the
+beginning of unexplored spaces.  Being able to play with various patterns in research
+without having to be an expert in how to modify Kubelet and it's multiple internal
+managers is a big benefit to the research community.
+
+#### User-specific plugins
+
+A user may have very specific allocation patterns they require.  This sort of capability
+may be rare and not belong upstream in mainstream Kubernetes, but there should still
+be a simple way to do allow users to do their specific experiments.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+#### Compute Specification Option
+
+The CCI model combined with some of the capabilities introduced in the Dynamic 
+Resource Allocation (DRA) KEP [3], offers the ability to transition compute 
+resource allocation behavior from being a cluster-admin dominated configuration
+to one that allows users with precise compute requirements to articulate the
+compute attributes that they need. 
```

Comment:
Beware that DRA is inherently going to be slower than the traditional resource assignments through kube-scheduler. Because kube-scheduler needs to communicate with external components, it cannot be as fast as doing everything in parallel in memory in a single process.

I am currently working on `test/integration/scheduler_perf` to measure how fast (or slow...) DRA really is and where things can still be optimized, therefore it is too early to discuss performance numbers.

Reply:
Thanks! CCI drivers should be able to coexist with DRA but also stand-alone.